### ISSUE TITLE
MDNS support

### ISSUE THREAD

**Issue by svenrademakers** [turing-machines/BMC-Firmware#20]
As a user, i want to be able to discover my BMC by a default hostname. for example turing_pi.local. This way I don't have to log in to the board to find or setup a static IP.

note: mdns discovery could be blocked by the users' local network. 

# Acceptance Criteria

* When the baseboard is booted the mdns service is started
* Im able to find and ping hostname "turingpi.local"
* When there are more turingpi boards on the network a counter will be added to the hostname. The last booted board has the highest counter value. e.g. "turingpi4.local"

**Comment by CFSworks** [comment]
:+1: This also works great in situations where the board cannot get an IP address (no DHCP server or RA daemon): it can instead assign link-local addresses (`fe80::...`+`169.254....`) and announce those via mDNS (and/or llmnr).

This could be good for out-of-box-experience too: for initial power-on, the user is encouraged to plug in a PSU and connect their PC directly to one of the two Ethernet ports. Then, they navigate a browser to `turingpi.local` where they are prompted to set up authentication, IP settings (DHCP vs. static, ...), and other key settings before they then connect their board to a real network.

**Comment by svenrademakers** [comment]
I like the idea of having a configuration page exposed over a link-local connection. I created a new issue for this #55. For any amendments or discussions please defer to ticket #55. This ticket we continue to use to track basic mdns support
